Phaethon/Tests/IntegrationTests/LoginTest.cs

```diff
@@ -0,0 +1,202 @@
+            using (var db = new DatabaseContext())
+            {
+                Login deleted = db.Login.Attach(userModel2);
+                loginDa.Delete(db, deleted);
+            }
+        }
+
+        #region Post
+
+        [Test]
+        public async Task Login_CorrectLoginInfo_OK()
+        {
+            //Setup
+            InitializeData();
+            userModel2.Password = OGPass;
+
+            var response = await _client.PostAsJsonAsync("Login/Login", userModel2);
+            
+            //Assert
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+
+            Cleanup();
+        }
+
+        [Test]
+        public async Task Login_NullPasswordLoginInfo_BadRequest()
+        {
+            //Setup
+            InitializeData();
+            
+            //Act
+            userModel2.Password = null;
+            var response = await _client.PostAsJsonAsync("Login/Login", userModel2);
+
+            //Assert
+            Assert.AreEqual(HttpStatusCode.BadRequest, response.StatusCode);
+
+            Cleanup();
+        }
+
+        [Test]
+        public async Task Login_NullUserNameLoginInfo_BadRequest()
+        {
+            //Setup
+            InitializeData();
+
+            //Act
+            userModel2.Username = null;
+            var response = await _client.PostAsJsonAsync("Login/Login", userModel2);
+
+            //Assert
+            Assert.AreEqual(HttpStatusCode.BadRequest, response.StatusCode);
+
+            Cleanup();
+        }
+        #endregion
+
+        #region Delete
+
+        [Test]
+        public async Task Delete_CorrectLoginInfo_OK()
+        {
+            // Setup
+           
+                InitializeData();
+           
+            //Act
+           
+            var response = await _client.PostAsJsonAsync("Login/Delete", userModel2.ID.ToString());
+
+            //Assert
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+        }
+        #endregion
+
+        #region Add
+        [Test]
+        public async Task New_User_OK()
+        {
+            // Setup
+            InitializeData();
+
+            //Act
+            using (var db = new DatabaseContext())
+            {
+                count1 = db.Login.Count();
+                db.Login.Add(userModel);
+                db.SaveChanges();
+            }
+            using (var db = new DatabaseContext())
+            {
+                count2 = db.Login.Count();               
+            }
+            var response = await _client.PostAsJsonAsync("Login/CreateOrUpdate", userModel);
+
+            //Assert
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+            Assert.AreNotEqual(count1, count2);
+            loginDa = new LoginDa { };
+            using (var db = new DatabaseContext())
+            {
+                Login deleted = db.Login.Attach(userModel);
+                loginDa.Delete(db, deleted);
+            }
+            Cleanup();
+        }
```

Comment:
Excuse me, but wtf is this test doing?
https://github.com/RaidenRabit/Phaethon/blob/da6d6b8edb567c6e375539f3d112132bbce69993/Phaethon/Tests/IntegrationTests/LoginTest.cs#L132-L158

Reply:
![image](https://user-images.githubusercontent.com/22789107/49926348-a7104180-febb-11e8-8854-5eeff41d015f.png)

Well, let me tell you something:
- of course the 2 counts are different, you just inserted 2 users in the db, by directly calling it. Which is fundamentally wrong, as this test is supposed to test if the API is able to insert a user, not if the db can insert user.
- Since Subject1 was inserted with DB query, and Username is supposed to unique, inserting Subject1, using the API should not be possible, since its already there. Meaning theres some API issue as well. (this case is particularly bad, because someone might just know the username of an account, and just create a new user, with the same name and simply replacing the previous one.